server/fixity.go

```diff
@@ -139,3 +155,118 @@ func (s *RESTServer) scanfixity() {
 	}
 	log.Println("Ending scanfixity. duration = ", time.Now().Sub(starttime))
 }
+
+func (s *RESTServer) GetFixityHandler(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+
+	item := r.FormValue("item")
+	start := r.FormValue("start")
+	end := r.FormValue("end")
+	status := r.FormValue("status")
+
+	log.Println("GetFixityHandler called")
+	log.Println("item= ", item)
+	log.Println("start= ", start)
+	log.Println("end= ", end)
+	log.Println("status= ", status)
+
+	startValue, startErr := startValidate(start)
+	if startErr != nil {
+		w.WriteHeader(400)
+		log.Println("GetFixityHandler :", startErr.Error())
+		return
+	}
+
+	endValue, endErr := endValidate(end)
+	if endErr != nil {
+		w.WriteHeader(400)
+		log.Println("GetFixityHandler :", endErr.Error())
+		return
+	}
+
+	itemValue, itemErr := itemValidate(item)
+	if itemErr != nil {
+		w.WriteHeader(400)
+		log.Println("GetFixityHandler :", itemErr.Error())
+		return
+	}
+
+	statusValue, statusErr := statusValidate(status)
+	if statusErr != nil {
+		w.WriteHeader(400)
+		log.Println("GetFixityHandler :", statusErr.Error())
+		return
+	}
+
+	fixityResults := s.Fixity.GetFixity(startValue, endValue, itemValue, statusValue)
+	if fixityResults == nil {
+		w.WriteHeader(404)
+		log.Println("GetFixityHandler start =", startValue, "end= ", endValue, "item =", itemValue, "status = ", statusValue, " Returns 404")
+		return
+	}
+
+	//Return results
+	enc := json.NewEncoder(w)
+	enc.Encode(fixityResults)
+}
+
+func (s *RESTServer) GetFixityIdHandler(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	id := ps.ByName("id")
+	log.Println("GetFixityIdHandler passed id =", id)
+	thisFixity := s.Fixity.GetFixityById(id)
+
+	if thisFixity == nil {
+		w.WriteHeader(404)
+		log.Println("GetFixityIdHandler id =", id, " Returns 404")
+		return
+	}
+
+	enc := json.NewEncoder(w)
+	enc.Encode(thisFixity)
+}
+
+// Some validation routines for GET /fixity params
+
+func startValidate(param string) (string, error) {
```

Comment:
See suggestion above to use `time.Time` type instead of a string.

Reply:
Same answer as above.